crates/engine/tree/src/tree/mod.rs

```diff
@@ -856,6 +1077,12 @@ where
                         self.process_payload_attributes(attr, &canonical_header, state, version);
                     return Ok(TreeOutcome::new(updated))
                 }
+
+                // Update the latest block state to reflect the canonical ancestor.
+                // This ensures that state providers and the transaction pool operate with
+                // the correct chain state after forkchoice update processing.
+                self.update_latest_block_to_canonical_ancestor(&canonical_header)?;
```

Comment:
Is it correct that this is being called after the `process_payload_attributes` if-branch? If `attrs.is_some()` then this line never gets hit.